games-engines/fheroes2/fheroes2-1.1.6.ebuild

```diff
@@ -0,0 +1,81 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
+inherit cmake python-single-r1 xdg
+
+DESCRIPTION="Recreation of HoMM2 game engine"
+HOMEPAGE="https://ihhub.github.io/fheroes2/"
+SRC_URI="https://github.com/ihhub/fheroes2/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64"
+IUSE="tools"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="
+	media-libs/libpng:=
+	media-libs/libsdl2[video]
+	media-libs/sdl2-image
+	media-libs/sdl2-mixer
+	sys-libs/zlib
+"
+RDEPEND="
+	${DEPEND}
+	${PYTHON_DEPS}
+	app-arch/libarchive
+	virtual/libintl
+"
+BDEPEND="
+	sys-devel/gettext
+	virtual/pkgconfig
+"
+
+PATCHES=(
+	"${FILESDIR}/fheroes2-1.0.4-scripts.patch"
+)
+
+src_prepare() {
+	sed -i -e "s/python3/${EPYTHON}/" script/homm2/extract_homm2_resources.sh || die
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	# Not using dev-games/libsmacker because the game crashes with it
+	local mycmakeargs=(
+		-DENABLE_IMAGE=ON
+		-DENABLE_TOOLS=$(usex tools)
+	)
+	cmake_src_configure
+}
+
+src_compile() {
+	cmake_src_compile
+	emake -C files/lang
+}
+
+src_install() {
+	cmake_src_install
+
+	if use tools; then
+		for file in 82m2wav bin2txt extractor h2dmgr icn2img pal2img til2img xmi2midi; do
```

Comment:
I'll add this on merging, but for these, make sure to `local file` above the loop (or elsewhere in the function if you must) to avoid it escaping scope and leaking

Reply:
ah, interesting, ok